### ISSUE TITLE
NoMethodError: [!] undefined method `real_path' for nil:NilClass

### ISSUE THREAD

**Issue by StevenMasini** [fastlane/fastlane#13667]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

`automatic_code_signing` started to failed last week for no obvious reason.
Initially I was on fastlane `2.103.1` but even after updating to `2.108.0` I get the same error.

Also if I try to comment `automatic_code_signing` to skip this action, but the next action fail with the same error.

##### Complete output when running fastlane, including the stack trace and command used
<details>
  <pre>bundler: failed to load command: fastlane (/Users/danielhu/.rbenv/versions/2.3.3/bin/fastlane)
NoMethodError: [!] undefined method `real_path' for nil:NilClass
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/xcodeproj-1.7.0/lib/xcodeproj/project/object/container_item_proxy.rb:96:in `container_portal_annotation'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/xcodeproj-1.7.0/lib/xcodeproj/project/object/container_item_proxy.rb:105:in `to_hash_as'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/xcodeproj-1.7.0/lib/xcodeproj/project/object.rb:421:in `to_ascii_plist'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/xcodeproj-1.7.0/lib/xcodeproj/project.rb:293:in `block in to_ascii_plist'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/xcodeproj-1.7.0/lib/xcodeproj/project.rb:291:in `each'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/xcodeproj-1.7.0/lib/xcodeproj/project.rb:291:in `to_ascii_plist'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/xcodeproj-1.7.0/lib/xcodeproj/project.rb:360:in `block in save'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/atomos-0.1.3/lib/atomos.rb:21:in `block in atomic_write'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/2.3.0/tempfile.rb:295:in `open'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/atomos-0.1.3/lib/atomos.rb:17:in `atomic_write'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/xcodeproj-1.7.0/lib/xcodeproj/project.rb:359:in `save'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/actions/automatic_code_signing.rb:69:in `run'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/fast_file.rb:149:in `method_missing'
  Fastfile:53:in `block (2 levels) in parsing_binding'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/lane.rb:33:in `call'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:204:in `try_switch_to_lane'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:146:in `trigger_action_by_name'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/fast_file.rb:149:in `method_missing'
  Fastfile:89:in `block (2 levels) in parsing_binding'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/lane.rb:33:in `call'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
  /Users/danielhu/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.108.0/bin/fastlane:23:in `<top (required)>'
  /Users/danielhu/.rbenv/versions/2.3.3/bin/fastlane:23:in `load'
  /Users/danielhu/.rbenv/versions/2.3.3/bin/fastlane:23:in `<top (required)>'
make: *** [deploy-ios-dev] Error 1</pre>
</details>

### Environment

<details>
  <pre><details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.14                                       |
| Ruby                        | 2.3.3                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.15.2 (Apple Git-101.1)        |
| Installation Source         | ~/.rbenv/versions/2.3.3/bin/fastlane        |
| Host                        | Mac OS X 10.14 (18A391)                     |
| Ruby Lib Dir                | ~/.rbenv/versions/2.3.3/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2p  14 Aug 2018                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.4.1                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:ios)

platform :ios do
    before_all do |lane, options|
        # Setup the slack URL webhook
        ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
    end

    desc "bundle and upload mapping files to Bugsnag server"
    lane :bundle do |options|
      BUILD_VERSION_NAME = options[:version]
      Dir.chdir("../..") do
        # generate bundle and mapping file
        sh "react-native bundle \
            --platform ios \
            --dev false \
            --entry-file index.js \
            --bundle-output ios/index.ios.bundle \
            --sourcemap-output ios.bundle.map"
        # upload mapping file to 
        sh "bugsnag-sourcemaps upload \
            --api-key '...' \
            --app-version #{BUILD_VERSION_NAME} \
            --minified-file ios/index.ios.bundle \
            --source-map ./ios.bundle.map \
            --minified-url main.jsbundle \
            --upload-sources"
      end
    end

    lane :refresh_dsyms do |options|
        download_dsyms(version: options[:version])  # Download dSYM files from iTC
        # https://github.com/bugsnag/bugsnag-dsym-upload/issues/24
        upload_symbols_to_bugsnag(dsym_path: [Actions.lane_context[SharedValues::DSYM_OUTPUT_PATH]]) # Upload them to Bugsnag
        clean_build_artifacts                       # Delete the local dSYM file
    end
  
    desc "Setup the build"
    lane :setup_build do |options| 
        
        # enable automatic code signing
        enable_automatic_code_signing(path: "Select.xcodeproj")
        
        # retrieve Info.plist path
        plist_path = File.expand_path '../Select/Info.plist'

        # set app version number
        set_info_plist_value(path: plist_path, key: "CFBundleShortVersionString", value: options[:version])

        # set app build number
        set_info_plist_value(path: plist_path, key: "CFBundleVersion", value: options[:build])

        # set facebook app id
        set_info_plist_value(path: plist_path, key: "FacebookAppID", value: options[:facebook])

        # set facebook uri schemes
        # Advanced processing: find URL scheme for particular key and replace value
        update_info_plist(
            scheme: "Select",
            block: proc do |plist|
                urlScheme = plist["CFBundleURLTypes"].find{|scheme| scheme["CFBundleURLName"] == "co.s12e.one"}
                urlScheme[:CFBundleURLSchemes] = ["home", "fb" + options[:facebook]]
            end
            )
    end

    desc "Build and deploy a new SELECT for Developers archive to Testflight"
    lane :deploy_dev do |options|
        if options.has_key?(:version) && options.has_key?(:build) && options.has_key?(:facebook)
            # Attach the badge to the App Icon
            shield = 'v' + options[:version] + '-' + options[:build] + '-orange'
            add_badge(shield: shield, no_badge: true, shield_gravity: 'South')
            
            # bundle and upload mapping files to Bugsnag server
            bundle(version: options[:version])

            # setup the build
            setup_build(options)

            # install or update cocoapods
            cocoapods
            
            # update bundle identifier and display name
            update_app_identifier(xcodeproj: "Select.xcodeproj", plist_path: "Select/Info.plist", app_identifier: "...");
            set_info_plist_value(path: "Select/Info.plist", key: "CFBundleDisplayName", value: "SELECT DEV");
            
            # build the archive
            build_app(scheme: "Select", include_bitcode: true)
        
            # Upload IPA to Testflight
            upload_to_testflight(skip_waiting_for_build_processing: false, app_identifier: "...")

            # Download the dSym file and upload it to Bugsnag
            refresh_dsyms(options)

            # Send a confirmation message to the Slack channel
            slackMessage = "Successfully delivered SELECT for Developers v" + options[:version] + " - " + options[:build] + " to Testflight."
            slack(message: slackMessage)
        else
            UI.user_error!("Error Missing parameter: e.g fastlane deploy_dev version:1.33.1 build: 13301 facebook:...")
        end 
    end

    desc "Build and deploy a new SELECT for Staging archive to Testflight"
    lane :deploy_stg do |options|
        if options.has_key?(:version) && options.has_key?(:build) && options.has_key?(:facebook)
            # Attach the badge to the App Icon
            shield = 'v' + options[:version] + '-' + options[:build] + '-blue'
            add_badge(shield: shield, no_badge: true, shield_gravity: 'South')

            # bundle and upload mapping files to Bugsnag server
            bundle(version: options[:version])

            # setup the build
            setup_build(options)

            # install or update cocoapods
            cocoapods

            # update bundle identifier and display name 
            update_app_identifier(xcodeproj: "Select.xcodeproj", plist_path: "Select/Info.plist", app_identifier: "...");
            set_info_plist_value(path: "Select/Info.plist", key: "CFBundleDisplayName", value: "SELECT STG");
            
            # build the archive
            build_app(scheme: "Select", include_bitcode: true)
        
            # Upload IPA to Testflight
            upload_to_testflight(skip_waiting_for_build_processing: false, app_identifier: "...")

            # Download the dSym file and upload it to Bugsnag
            refresh_dsyms(options)

            # Send a confirmation message to the Slack channel
            slackMessage = "Successfully delivered SELECT for Staging v" + options[:version] + " - " + options[:build] + " to Testflight."
            slack(message: slackMessage)
        else
            UI.user_error!("Error Missing parameter: e.g fastlane deploy_stg version:1.33.1 build: 13301 facebook:...")
        end 
    end

    desc "Build and deploy a new IPA to the App Store"
    lane :deploy_prod do |options|
        if options.has_key?(:version) && options.has_key?(:build) && options.has_key?(:facebook)
            # setup the build
            setup_build(options)

            # install or update cocoapods
            cocoapods
            
            # update bundle identifier and display name
            update_app_identifier(xcodeproj: "Select.xcodeproj", plist_path: "Select/Info.plist", app_identifier: "...");
            set_info_plist_value(path: "Select/Info.plist", key: "CFBundleDisplayName", value: "SELECT");

            # build the archive
            build_app(scheme: "Select", include_bitcode: true)
        
            # Upload IPA to Testflight
            upload_to_testflight(skip_waiting_for_build_processing: false, app_identifier: "...")

            # Download the dSym file and upload it to Bugsnag
            refresh_dsyms(options)

            # Send a confirmation message to the Slack channel
            slackMessage = "Successfully delivered SELECT v" + options[:version] + " - " + options[:build] + " to the App Store."
            slack(message: slackMessage)
        else
            UI.user_error!("Error Missing parameter: e.g fastlane deploy_prod version:1.33.1 build: 13301 facebook:...")
        end 
    end

    error do |lane, exception|
        slack(message: exception.message, success: false)
    end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "..." # The bundle identifier of your app
apple_id "..." # Your Apple email address

itc_team_id "..." # iTunes Connect Team ID
team_id "V2HPRJ6JEE" # Developer Portal Team ID

# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.108.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                  | Version | Update-Status |
| ----------------------- | ------- | ------------- |
| fastlane-plugin-bugsnag | 1.4.0   | ✅ Up-To-Date  |
| fastlane-plugin-badge   | 1.1.0   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| bundler                   | 1.17.1       |
| CFPropertyList            | 3.0.0        |
| concurrent-ruby           | 1.0.5        |
| i18n                      | 0.9.5        |
| minitest                  | 5.11.3       |
| thread_safe               | 0.3.6        |
| tzinfo                    | 1.2.5        |
| activesupport             | 4.2.10       |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| atomos                    | 0.1.3        |
| babosa                    | 1.0.2        |
| curb                      | 0.9.6        |
| fastimage                 | 2.1.4        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| dotenv                    | 2.5.0        |
| emoji_regex               | 0.1.1        |
| excon                     | 0.62.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.15.3       |
| unf_ext                   | 0.0.7.5      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20180417 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.12.2       |
| gh_inspector              | 1.1.3        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| multi_json                | 1.13.1       |
| os                        | 1.0.0        |
| signet                    | 0.11.0       |
| googleauth                | 0.6.7        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2018.0812  |
| mime-types                | 3.2.2        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.23.9       |
| json                      | 2.1.0        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.4.0        |
| rubyzip                   | 1.2.2        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.5        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.4.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.6.5        |
| tty-cursor                | 0.6.0        |
| tty-spinner               | 0.8.0        |
| word_wrap                 | 1.0.0        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.7.0        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 1.0.0        |
| badge                     | 0.9.1        |
| fuzzy_match               | 2.0.4        |
| nap                       | 1.1.0        |
| cocoapods-core            | 1.5.3        |
| cocoapods-deintegrate     | 1.0.2        |
| cocoapods-downloader      | 1.2.1        |
| cocoapods-plugins         | 1.0.0        |
| cocoapods-search          | 1.0.0        |
| cocoapods-stats           | 1.0.0        |
| netrc                     | 0.11.0       |
| cocoapods-trunk           | 1.3.0        |
| cocoapods-try             | 1.1.0        |
| escape                    | 0.0.4        |
| fourflusher               | 2.0.1        |
| molinillo                 | 0.6.5        |
| ruby-macho                | 1.2.0        |
| cocoapods                 | 1.5.3        |
| fastlane-plugin-badge     | 1.1.0        |
| git                       | 1.5.0        |
| xml-simple                | 1.1.5        |
| fastlane-plugin-bugsnag   | 1.4.0        |
</details>

*generated on:* **2018-11-12**
</details></pre>
</details>

**Comment by StevenMasini** [comment]
@janpio thank you for helping me out.

You are right, I tried to build a previous version of my project and it worked. But I can build the latest version of my project.

Actually right now we have a weird configuration in our company, my teammates and myself already upgrade to **XCode 10.1** whereas our build machine is still under **XCode 9.4.1**

Could it be due to this ?

Thank you for you help again.

**Comment by StevenMasini** [comment]
@janpio I just tried to build via Fastlane from my machine using **XCode 10.1** and I got the exact same error.

But you are right, this issue does not occur if I compile an older version of my project. 

Do you have any idea what can cause this error ?

**Comment by StevenMasini** [comment]
@janpio So it's not a problem relative to my version of XCode. I tried to compile the previous version of my project using both **XCode 10.1** and **XCode 9.4.1** and both worked without any error.

So most likely it has to see with something that I introduced.

**Comment by janpio** [comment]
Not really. Projects are just folders, so you can diff them. Might be useful to see what exactly changed.

**Comment by StevenMasini** [comment]
@janpio Ok I found the reason!

Nothing wrong with `xcodeproj` or `fastlane`.
What happened is that my `project.xcodeproj` file got corrupted by one of the library that I added.

Every time a library is added, XCode will reference this library in the `project.xcodeproj` file, with an object identifiable with the key `containerPortal`.

In my case I had  a library that was reference twice into my `project.xcodeproj`, but one of the reference was dead. In order to fix this issue, I had to remove the library from my XCode project, and then dig into my `project.xcodeproj` to manually remove all the dead reference of this library. Then finally add back the library via XCode.

Hope my explanation is clear enough, in case someone else encounter the same issue as me.

Thank you @janpio
